crates/uplc/src/builder.rs

```diff
@@ -2,27 +2,21 @@ use crate::{
     ast::{Constant, Name, Term, Type},
     builtins::DefaultFunction,
 };
+use pallas_primitives::alonzo::PlutusData;
 
 pub const CONSTR_FIELDS_EXPOSER: &str = "__constr_fields_exposer";
 pub const CONSTR_INDEX_EXPOSER: &str = "__constr_index_exposer";
 pub const CONSTR_GET_FIELD: &str = "__constr_get_field";
 pub const EXPECT_ON_LIST: &str = "__expect_on_list";
 
-impl Term<Name> {
+impl<T> Term<T> {
```

Comment:
I had to slightly generalize that one to make it reusable with `Term<DeBruijn>` as well; so I've split methods between `impl<T> Term<T>` when applicable and `impl Term<Name>` for those that requires actual named variables.

Reply:
Nice looks cool, thanks